collections/src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -575,6 +691,100 @@ sealed abstract class Stream[+A] extends LinearSeq[A] with LazyListOps[A, Stream
     if (this.isEmpty) z
     else tail.foldLeft(op(z, head))(op)
   }
+
+
+  protected def tailDefined: Boolean
+
+  override def toString: String = {
```

Comment:
Can this implementation be shared between LazyList and Stream?

Reply:
I thought this wasn’t useful since `Stream` is deprecated anyway.

Reply:
OK, I’ve factored the implementation out in `LazyListOps`.